### PR TITLE
Course list improvements

### PR DESCRIPTION
**Pull request by bjrne**

### CODE REVIEW

File: xikolo-ios/ViewControllers/Courses/CourseListViewController.swift [hunk 1]
```diff
@@ -31,7 +33,21 @@ class CourseListViewController : AbstractCourseListViewController {
...
+            numberOfItemsPerRow = 2
+        }
+    }
+
+    override func traitCollectionDidChange(previousTraitCollection: UITraitCollection?) {
+        detectSizeClass()
+    }
+
+    override func viewWillTransitionToSize(size: CGSize, withTransitionCoordinator coordinator:
+        UIViewControllerTransitionCoordinator) {
```
Comment (invliD) [inline xikolo-ios/ViewControllers/Courses/CourseListViewController.swift]: Please remove the linebreak.

File: xikolo-ios/ViewControllers/Courses/CourseListViewController.swift [hunk 2]
```diff
@@ -31,7 +33,21 @@ class CourseListViewController : AbstractCourseListViewController {
...
+        switch traitCollection.horizontalSizeClass {
+        case UIUserInterfaceSizeClass.Compact, UIUserInterfaceSizeClass.Unspecified:
+            numberOfItemsPerRow = 1
+        case UIUserInterfaceSizeClass.Regular:
+            numberOfItemsPerRow = 2
+        }
+    }
+
+    override func traitCollectionDidChange(previousTraitCollection: UITraitCollection?) {
+        detectSizeClass()
```
Comment (invliD) [inline xikolo-ios/ViewControllers/Courses/CourseListViewController.swift]: Why move the code to a separate function? It's only called once and `traitCollectionDidChange` only calls that function. Just leave the code here.

File: xikolo-ios/Base.lproj/Localizable.strings [hunk 3]
```diff
@@ -27,6 +27,11 @@
 "password" = "Password";
 "register" = "Register";
 
+//Courses
+"status_active" = "active";
+"status_self-paced" = "self-paced";
+"status_announced" = "upcoming";
```
Comment (invliD) [inline xikolo-ios/Base.lproj/Localizable.strings]: Please remove these and use the English string directly in code.

File: xikolo-common/Data/Model/Course.swift [hunk 4]
```diff
@@ -80,6 +84,10 @@ class CourseSpine : BaseModelSpine {
             "language": Attribute(),
             "start_at": DateAttribute(),
             "end_at": DateAttribute(),
+            "status": Attribute(),
+            "hidden": Attribute(),
+            "open": Attribute(),
+            "external": Attribute(),
```
Comment (invliD) [inline xikolo-common/Data/Model/Course.swift]: `hidden`, `open` and `external` are bools. Please use `xx_int` and add a boolean setter/getter.

File: xikolo-ios/Views/CourseCell.swift [hunk 5]
```diff
@@ -23,7 +25,24 @@ class CourseCell : UICollectionViewCell {
...
         nameLabel.text = course.title
         teacherLabel.text = course.teachers
-        dateLabel.text = course.language
+        dateLabel.text = course.language_translated
+
+        switch course.status {
+        case "active"?:
+            statusView.hidden = false
+            statusLabel.text = NSLocalizedString("running", comment: "course-status")
+            statusView.backgroundColor = UIColor.init(red: 140/255, green: 179/255, blue: 13/255, alpha: 1.0)
```
Comment (invliD) [inline xikolo-ios/Views/CourseCell.swift]: Can you create a generic Brand file like in [62df1e4](https://github.com/openHPI/xikolo-ios/blob/62df1e4/xikolo-common/Branding/Brand%2Bgeneric.swift) on my branch and put these colors there?

File: xikolo-ios/ViewControllers/Courses/CourseListViewController.swift [hunk 6]
```diff
@@ -31,6 +33,15 @@ class CourseListViewController : AbstractCourseListViewController {
         updateView()
     }
 
+    override func traitCollectionDidChange(previousTraitCollection: UITraitCollection?) {
+        switch traitCollection.horizontalSizeClass {
+        case UIUserInterfaceSizeClass.Compact, UIUserInterfaceSizeClass.Unspecified:
```
Comment (invliD) [inline xikolo-ios/ViewControllers/Courses/CourseListViewController.swift]: Is `UIUserInterfaceSizeClass` required here, i.e. is `case .Compact, .Unspecified:` not enough?

File: xikolo-ios/Views/CourseCell.swift (hunk 5 above)
Comment (bjrne) [inline xikolo-ios/Views/CourseCell.swift]: I originally thought it gets merged long before mine :D

**Pull request merged**